### PR TITLE
Initial pipeline-execution resource

### PR DESCRIPTION
**Pull request by ryansteakley**
Description of changes:

- Initial pipeline execution resource
- E2E tests are a work in progress - ( Don't merge until they are done)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: apis/v1alpha1/ack-generate-metadata.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ack_generate_info:
-  build_date: "2022-10-04T18:55:06Z"
+  build_date: "2022-10-12T22:54:58Z"
   build_hash: 6e2ffbc3b16a30ac59be6719918c601c2c864064
-  go_version: go1.17.13
-  version: v0.20.1-3-g6e2ffbc
-api_directory_checksum: ca9187f53c674d6424c5a4120fe2609afce3d52a
+  go_version: go1.19.1
+  version: v0.20.1-3-g6e2ffbc-dirty
```
Comment (ananth102) [inline apis/v1alpha1/ack-generate-metadata.yaml:5]: was the code generator modified?

File: config/controller/kustomization.yaml [hunk 2]
```diff
@@ -5,5 +5,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: public.ecr.aws/aws-controllers-k8s/sagemaker-controller
-  newTag: v0.4.5
+  newName: 105083556692.dkr.ecr.us-east-1.amazonaws.com/sagemaker-controller
+  newTag: v0.5.0-rc.0
```
Comment (ananth102) [inline config/controller/kustomization.yaml:9]: Is this going to trigger a release?

File: pkg/testutil/test_suite_runner.go [hunk 3]
```diff
@@ -116,15 +116,19 @@ func (runner *TestSuiteRunner) runTestScenario(t *testing.T, scenarioName string
 	runner.assertExpectations(assert, expectation, actual, err)
 }
 
-/* assertExpectations validates the actual outcome against the expected outcome.
+/*
+	assertExpectations validates the actual outcome against the expected outcome.
```
Comment (ananth102) [inline pkg/testutil/test_suite_runner.go:120]: nit: any reason for this change? Looks off when viewing the file.

File: templates/pipeline_execution/sdk_delete_pre_build_request.go.tpl [hunk 4]
```diff
@@ -0,0 +1,12 @@
+	latestStatus := r.ko.Status.PipelineExecutionStatus
+	if latestStatus != nil {
+		if *latestStatus == svcsdk.PipelineExecutionStatusStopping {
+			return r, requeueWaitWhileDeleting
+		}
+
+		// Call StopPipelineExecution only if the job is Executing, otherwise just 
+		// return nil to mark the resource Unmanaged
```
Comment (ananth102) [inline templates/pipeline_execution/sdk_delete_pre_build_request.go.tpl:8]: Is this comment out of date?

File: test/e2e/resources/pipeline_execution.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+apiVersion: sagemaker.services.k8s.aws/v1alpha1
+kind: PipelineExecution
+metadata:
+  name: $PIPELINE_EXECUTION_RESOURCE_NAME
+spec:
+  pipelineExecutionDisplayName: $PIPELINE_EXECUTION_RESOURCE_NAME
+  pipelineName: serial-inference-pipeline
```
Comment (ananth102) [inline test/e2e/resources/pipeline_execution.yaml:7]: Is this created in the other test?

**Review by ananth102 — COMMENTED** [review]
Looks good, just a couple small things.

File: test/e2e/resources/pipeline_execution.yaml (hunk 5 above)
Comment (surajkota) [reply test/e2e/resources/pipeline_execution.yaml:7]: pipeline creation missing and use a smaller pipeline?

File: test/e2e/tests/test_pipeline_execution.py [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+        resource = k8s.patch_custom_resource(reference, spec)
+        resource = k8s.wait_resource_consumed_by_controller(reference)
+        assert resource is not None
+
+        self._assert_pipeline_execution_status_in_sync(
+            pipeline_execution_arn, reference, cfg.JOB_STATUS_EXECUTING
+        )
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False")
+
+        pipeline_execution_desc = get_sagemaker_pipeline_execution(
```
Comment (surajkota) [inline test/e2e/tests/test_pipeline_execution.py:171]: verify last modified time got updated

File: helm/Chart.yaml [hunk 7]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
 name: sagemaker-chart
 description: A Helm chart for the ACK service controller for Amazon SageMaker (SageMaker)
-version: v0.4.5
-appVersion: v0.4.5
+version: v0.5.0-rc.0
```
Comment (surajkota) [inline helm/Chart.yaml:4]: revert release related changes

**Review by surajkota — CHANGES_REQUESTED** [review]
Unit tests and I remember there was a problem with adopting the resource, is there a fix for it now?

File: test/e2e/tests/test_pipeline_execution.py (hunk 6 above)
Comment (ryansteakley) [reply test/e2e/tests/test_pipeline_execution.py:171]: sure with last-modified-time, afaik tags do not exist for this resource

File: templates/pipeline_execution/sdk_delete_pre_build_request.go.tpl (hunk 4 above)
Comment (ryansteakley) [reply templates/pipeline_execution/sdk_delete_pre_build_request.go.tpl:8]: Believe it is still valid, in the case where we want to delete a pipeline_execution and its status is Succeeded

File: pkg/testutil/test_suite_runner.go (hunk 3 above)
Comment (ryansteakley) [reply pkg/testutil/test_suite_runner.go:120]: some auto-formatting issue when I save using visual-studio, will edit

File: generator.yaml [hunk 8]
```diff
@@ -925,5 +983,6 @@ ignore:
       - InstanceGroups
```
Comment (ananth102) [inline generator.yaml:983]: Can you remove InstanceGroups and InstanceGroupNames and regenerate

Comment (ryansteakley) [reply generator.yaml:983]: reason?

Comment (ryansteakley) [reply generator.yaml]: done

File: pkg/testutil/test_suite_runner.go (hunk 3 above)
Comment (surajkota) [reply pkg/testutil/test_suite_runner.go:120]: does this need to be reverted?

**Review by surajkota — APPROVED** [review]
looks good, one nit

File: test/e2e/tests/test_pipeline_execution.py [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+        assert (
+            pipeline_execution_desc["PipelineExecutionDisplayName"]
+            == new_pipeline_execution_display_name
+        )
+        assert (
+            resource["spec"].get("pipelineExecutionDisplayName", None)
+            == new_pipeline_execution_display_name
+        )
+
+        self._assert_pipeline_execution_status_in_sync(
```
[the remark this answers is not in the document]
Comment (ryansteakley) [reply test/e2e/tests/test_pipeline_execution.py:227]: done

**Pull request merged**